sdks/go/pkg/beam/runners/dataflow/translate.go

```diff
@@ -290,18 +290,12 @@ func expandCoGBK(nodes map[int]*outputReference, edge *graph.MultiEdge) ([]*df.S
 	return steps, nil
 }
 
-// TODO(herohde) 5/16/2017: we might need to robustly encode the index into the
-// name, so that we can infer the ordering from the names in the harness.
-
 // translateNodes builds a map from nodeID to the Dataflow representation.
 func translateNodes(edges []*graph.MultiEdge) map[int]*outputReference {
 	nodes := make(map[int]*outputReference)
 	for _, edge := range edges {
 		for i, out := range edge.Output {
-			name := "out"
-			if i > 0 {
-				name = fmt.Sprintf("side%v", i)
-			}
+			name := fmt.Sprintf("i%v", i)
```

Comment:
Consider adding a comment to point to/reference where we rely on the input being i%d, since it's in the harness code.
eg.
// Output name format required by beam/runtime/harness.myUnexportedMethodNameWhateverItIs()

Reply:
Done.